app/src/operations_endpoints.h

```diff
@@ -0,0 +1,681 @@
+   */
+  class OperationsIndexingStrategy
+    : public VisitEachEntryInValueTyped<OperationsTable>
+  {
+  public:
+    OperationsIndexingStrategy(ccf::BaseEndpointRegistry& registry) :
+      OperationsIndexingStrategy(
+        [&registry](timespec& time) {
+          return registry.get_untrusted_host_time_v1(time);
+        },
+        [&registry](
+          ccf::View view, ccf::SeqNo seqno, ccf::TxStatus& tx_status) {
+          return registry.get_status_for_txid_v1(view, seqno, tx_status);
+        })
+    {}
+
+    OperationsIndexingStrategy(
+      std::function<ccf::ApiResult(timespec& time)> get_time,
+      std::function<ccf::ApiResult(
+        ccf::View view, ccf::SeqNo seqno, ccf::TxStatus& tx_status)>
+        get_status_for_txid) :
+      VisitEachEntryInValueTyped(OPERATIONS_TABLE),
+      get_time(get_time),
+      get_status_for_txid(get_status_for_txid)
+    {}
+
+    /**
+     * Get the list of all operations whose state is known by the indexing
+     * strategy.
+     *
+     * The function isn't intended to be used in normal situations, but can
+     * serve as a convenient debugging endpoint into the state of the indexing
+     * strategy.
+     */
+    std::vector<GetOperation::Out> operations() const
+    {
+      std::lock_guard guard(lock);
+      std::vector<GetOperation::Out> result;
+      for (const auto& it : operations_)
+      {
+        result.push_back(GetOperation::Out{
+          .operation_id = ccf::TxID{it.second.view, it.first},
+          .status = it.second.status,
+          .entry_id = it.second.completion_tx,
+          .error = it.second.error,
+        });
+      }
+      return result;
+    }
+
+    /**
+     * Look up an operation by its transaction ID.
+     *
+     * If the transaction ID is unknown, this function generally returns an
+     * operation in the "running" state anyway, since it would not be possible
+     * to distinguish between whether this is an invalid transaction ID or if
+     * the transaction that created the operation has not been globally
+     * committed and indexed yet.
+     *
+     * In some cases however, we can be confident that this transaction ID is
+     * invalid now and forever in the future. In these cases, an appropriate
+     * HTTPError exception is thrown.
+     */
+    GetOperation::Out lookup(const ccf::TxID& operation_id) const
+    {
+      std::lock_guard guard(lock);
+
+      ccf::TxStatus tx_status;
+      auto result =
+        get_status_for_txid(operation_id.view, operation_id.seqno, tx_status);
+      if (result != ccf::ApiResult::OK)
+      {
+        throw InternalError(fmt::format(
+          "Failed to get transaction status: {}",
+          ccf::api_result_to_str(result)));
+      }
+
+      switch (tx_status)
+      {
+        case ccf::TxStatus::Unknown:
+        case ccf::TxStatus::Pending:
+          return {
+            .operation_id = operation_id,
+            .status = OperationStatus::Running,
```

Comment:
Note to self: if we stick with "running", update spec and emulator which currently use "pending".

Reply:
Done. Renamed to "running".